lib/connection/bind_uploader.js

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2015-2021 Snowflake Computing Inc. All rights reserved.
+ */
+ var Logger = require('../logger');
+
+const Readable = require('stream').Readable;
+const fs = require('fs');
+const tmp = require('tmp');
+
+var Statement = require('./statement');
+var fileCompressionType = require('.././file_transfer_agent/file_compression_type');
+
+
+const STAGE_NAME = 'SYSTEM$BIND';
+const CREATE_STAGE_STMT = "CREATE OR REPLACE TEMPORARY STAGE "
+	+ STAGE_NAME
+	+ " file_format=( type=csv field_optionally_enclosed_by='\"')";
+	
+/**
+ * Creates a new BindUploader.
+ *
+ * @param {Object} options
+ * @param {Object} services
+ * @param {Object} connectionConfig
+ * @param {*} requestId 
+ *
+ * @constructor
+ */
+ 
+function BindUploader(options, services, connectionConfig, requestId)
+{
+	const MAX_BUFFER_SIZE = 1024 * 1024 * 10;
+	
+	Logger.getInstance().debug('BindUploaders');
+	this.options = options;
+	this.services = services;
+	this.connectionConfig = connectionConfig;
+	this.requestId = requestId;
+	this.stagePath = '@' + STAGE_NAME + '/' + requestId;
+	Logger.getInstance().debug('token = %s', connectionConfig.getToken());
+
+	this.tempStageCreated = false;
+	this.bindData = null;
+	this.files = [];
+	this.datas = [];
+	this.puts = [];
+
+	this.Upload = function(bindings)
+	{
+		Logger.getInstance().debug('BindUploaders::Upload');
+	
+		if(bindings == null)
+			return null;
+	
+		var dataRows = new Array();
+		for(var i=0; i<bindings.length; i++)
+		{
+			var strbuffer = "";
+			for(var j=0; j< bindings[i].length; j++)
+			{
+				if (j>0)
+					strbuffer += ',';
+				var value = bindings[i][j];
+				strbuffer += value;
+			}
+			strbuffer += '\n';
+			dataRows.push(strbuffer);
+		}
+	
+		var startIndex = 0;
+		var rowNum = 0;
+		var curBytes = 0;
+		var fileCount = 0;
+		
+	
+		while (rowNum < dataRows.length)
+		{
+			Logger.getInstance().debug('rowNum='+rowNum+', dataRows='+dataRows.length);
```

Comment:
Can this entire loop not be done as part of the previous for loop where we're creating the CSV? Can the strbuffer created in the previous loop be ingested into data directly which is being done below?